### PR TITLE
Fix docker multi stage builds for alpine linux

### PR DESCRIPTION
**Pull request by paololim**

### CODE REVIEW

File: play/Dockerfile.play [hunk 1]
```diff
@@ -4,5 +4,10 @@ MAINTAINER tech@flow.io
 
 WORKDIR /root
 
+# install stuff
+RUN apk add --no-cache --update bash && \
+    apk add --no-cache --virtual=build-dependencies curl && \
```
Comment (benwaffle) [inline play/Dockerfile.play]: why not just `apk add curl`? also we don't need curl at runtime

File: play/Dockerfile.play_builder [hunk 2]
```diff
@@ -7,12 +7,9 @@ ENV SBT_VERSION 1.1.6
 WORKDIR /root
 
 # Install SBT
-RUN apk add --no-cache --update bash && \
-    apk add --no-cache --virtual=build-dependencies curl && \
```
Comment (benwaffle) [inline play/Dockerfile.play_builder]: we need to install curl to install sbt below

File: play/Dockerfile.play [hunk 3]
```diff
@@ -4,5 +4,9 @@ MAINTAINER tech@flow.io
 
 WORKDIR /root
 
+# install stuff
+RUN apk add --no-cache bash && \
```
Comment (grahamar) [inline play/Dockerfile.play]: You can do this in a single command

`RUN apk add --no-cache bash curl`

File: play/Dockerfile.play (hunk 1 above)
Comment (grahamar) [reply play/Dockerfile.play]: The reason to use the virtual flag here is that `build-dependencies` is only needed when installing curl but not for running curl, this is an optimisation step, best practice is to keep Docker images to purely what is required.

File: play/Dockerfile.play_builder (hunk 2 above)
Comment (grahamar) [reply play/Dockerfile.play_builder]: SBT is not installed in this image, play_builder no longer extends this image due to JRE vs JDK

File: play/Dockerfile.play (hunk 1 above)
Comment (benwaffle) [reply play/Dockerfile.play]: I believe --virtual just creates a virtual package of all the things you installed. It just makes it easy to uninstall everything later

Comment (grahamar) [reply play/Dockerfile.play]: Ah ok I misunderstood the manual then :)

**Pull request merged**